Add option to guess and calculate width and height for an image format (#36)
sulu/web-twig: docs/image.md

    '(max-width: 650px)': 'sulu-100x100',
}, { webp: 'image/webp' }) }}
```

##### 8. Set width and height attribute automatically

> Since Sulu 2.3 the original image width and height are save as properties.

Reviewer: ```suggestion
Since Sulu 2.3 the original image width and height are saves as properties.
```

Reviewer 2: `... are saved as properties.`

> This allows us the guess the width and height dimension of a given image format.

Reviewer: ```suggestion
This allows to guess the width and height of a image format and set the respective HTML attributes.
```

With setting the width and height attribute modern browsers avoid a layer shifts
and the page will not jump when images are loaded.

This feature can be activated the following way:

```yaml
services:
    Sulu\Twig\Extensions\ImageExtension:
        arguments:
            $guessAspectRatio: true
```

> So if we have an original image of 1920x1080 and have a image format called 100x:

Reviewer: ```suggestion
For example, if the original image has a resolution of 1920x1080 and the image format is called 100x:
```

```twig
{{ get_image(headerImage, '100x') }}
```

> Will add automatically the width and height attributes to the imgage tag:

Reviewer: ```suggestion
The feature will automatically add a width and height attribute to the rendered image tag:
```

```twig
<img alt="Title" title="Description" src="/uploads/media/100x/01/image.jpg?v=1-0" width="'100" height="56">
```

> This only works when writing image formats in specific format 100x, x100, 100x100, 100x100-inset, 100x@2x, ...
> for all other cases this option not be activated.

Reviewer: ```suggestion
This feature works only for image formats with keys that match the following formats: 100x, x100, 100x100, 100x100-inset, 100x@2x, ...
For other keys, the feature is not able to determine the dimensions and therefore cannot add a width and height attribute.
```

Author: changed this this paragraph with the new parameter

    '(max-width: 650px)': 'sulu-100x100',
}, { webp: 'image/webp' }) }}
```

##### 8. Set width and height attribute automatically

Since Sulu 2.3 the original image width and height are saved as properties.
This allows to guess the width and height of a image format and set the respective HTML attributes.

Setting the width and height attribute allows modern browsers to avoid layer shifts
and therefore the page will not jump when images are loaded.

This feature can be activated the following way:

```yaml
services:
    Sulu\Twig\Extensions\ImageExtension:
        arguments:
            $aspectRatio: true
            $imageFormatConfiguration: '%sulu_media.image.formats%' # optional but recommended
```

For example, if the original image has a resolution of 1920x1080 and the image format is called 100x:

```twig
{{ get_image(headerImage, '100x') }}
```

The feature will automatically add a width and height attribute to the rendered image tag:

```twig
<img alt="Title" title="Description" src="/uploads/media/100x/01/image.jpg?v=1-0" width="'100" height="56">
```

> The `$imageFormatConfiguration` parameter is optional. Without it will try to
> guess the format by the given format key. It will then only work with format keys
> like 100x, x100, 100x@2x, 100x100-inset and similars.

Reviewer: ```suggestion
The `$imageFormatConfiguration` parameter is optional. If it is not set, the extension
tries to guess the dimensions by the given format key. This will only work for format keys
in the format of 100x, x100, 100x@2x and 100x100-inset.
```
